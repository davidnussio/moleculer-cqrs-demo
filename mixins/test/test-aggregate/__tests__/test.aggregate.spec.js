const CQRSFixture = require("../../../../aggregates/cqrs-fixture");

const aggregate = require("../../test-aggregate");

const {
  commands: { createTest, deleteTest },
  events: { TestCreatedEvent, TestDeletedEvent },
} = aggregate;

const payload = {
  title: "Title test",
  userId: "user-id-1",
  text: "Asperiores nam tempora qui et provident temporibus illo et fugit.",
};

jest
  .spyOn(global.Date, "now")
  .mockImplementation(() => new Date("2019-05-14T11:01:58.135Z").valueOf());

describe("Testing aggregate commands in isolation", () => {
  test("should commands with empty payload throw error", () => {
    expect(() => createTest({}, {})).toThrow("Aggregate validation error");
  });

  test("should createTest command return an TestCreatedEvent", () => {
    expect(
      createTest(
        {},
        {
          payload,
        }
      )
    ).toMatchSnapshot();
  });
});

describe("Testing news aggregate with cqrs fixture", () => {
  let fixture;

  beforeEach(() => {
    fixture = new CQRSFixture(aggregate);
  });

  test("should call raw command", () => {
    fixture
      .givenEvents([])
      .when({
        aggregateId: "12345",
        aggregateName: "test",
        type: "createTest",
        payload,
      })
      .expectEvents(TestCreatedEvent(payload));
  });

  test("should command function", () => {
    fixture
      .givenEvents()
      .when(createTest, payload)
      .expectEvents(TestCreatedEvent(payload));
  });

  test("should reject upvode command when news already deleted", () => {
    fixture
      .givenEvents([TestCreatedEvent(payload), TestDeletedEvent()])
      .whenThrow(deleteTest, {})
      .toThrow("Aggregate is already deleted");
  });
});
