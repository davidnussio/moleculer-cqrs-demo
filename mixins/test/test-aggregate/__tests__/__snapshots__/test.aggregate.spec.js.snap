// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing aggregate commands in isolation should createTest command return an TestCreatedEvent 1`] = `
Object {
  "payload": Object {
    "text": "Asperiores nam tempora qui et provident temporibus illo et fugit.",
    "title": "Title test",
    "userId": "user-id-1",
  },
  "type": "aggregate.test.created",
}
`;
